### PR TITLE
Add CI test in single precision

### PR DESCRIPTION
**Pull request by lucafedeli88**
Since the issue fixed in https://github.com/ECP-WarpX/WarpX/pull/3954 was not found in CI, this PR adds a single precision CI compilation test that would have caught the issue.

✔️ https://github.com/ECP-WarpX/WarpX/pull/3954 should be merged before merging this PR.

### CODE REVIEW

File: .github/workflows/ubuntu.yml [hunk 1]
```diff
@@ -80,6 +80,46 @@ jobs:
         ./build/bin/warpx.1d Examples/Physics_applications/laser_acceleration/inputs_1d
         ./build/bin/warpx.2d Examples/Physics_applications/laser_acceleration/inputs_2d
 
+  build_3D_sp:
+    name: GCC 3D w/ MPI, single precision
```
Comment (ax3l) [inline .github/workflows/ubuntu.yml]: Shall we make this 3D + RZ?

File: .github/workflows/ubuntu.yml [hunk 2]
```diff
@@ -80,6 +80,46 @@ jobs:
...
+        key: ccache-openmp-3D-SP-${{ hashFiles('.github/workflows/ubuntu.yml') }}-${{ hashFiles('cmake/dependencies/AMReX.cmake') }}
+        restore-keys: |
+          ccache-openmp-3D-SP-${{ hashFiles('.github/workflows/ubuntu.yml') }}-
+          ccache-openmp-3D-SP-
+    - name: build WarpX
+      run: |
+        cmake -S . -B build                 \
+          -GNinja                           \
+          -DCMAKE_VERBOSE_MAKEFILE=ON       \
+          -DWarpX_DIMS="3"                  \
```
Comment (ax3l) [inline .github/workflows/ubuntu.yml]:
```suggestion
          -DWarpX_DIMS="RZ;3"                  \
```

File: .github/workflows/ubuntu.yml [hunk 3]
```diff
@@ -80,6 +80,46 @@ jobs:
...
+    if: github.event.pull_request.draft == false
+    env:
+      CXXFLAGS: "-Werror"
+      CXX: "g++-12"
+      CC: "gcc-12"
+    steps:
+    - uses: actions/checkout@v3
+    - name: install dependencies
+      run: |
+        .github/workflows/dependencies/gcc12.sh
```
Comment (ax3l) [inline .github/workflows/ubuntu.yml]: BLAS++ and LAPACK++ would be needed in this file then (or a new one)

File: .github/workflows/ubuntu.yml (hunk 2 above)
Comment (lucafedeli88) [reply .github/workflows/ubuntu.yml]: sure!

**Review by ax3l — APPROVED** [review]
Thank you! :+1:

File: .github/workflows/ubuntu.yml [hunk 4]
```diff
@@ -80,6 +80,45 @@ jobs:
...
+          -DCMAKE_VERBOSE_MAKEFILE=ON       \
+          -DWarpX_DIMS="RZ;3"                  \
+          -DWarpX_EB=OFF                    \
+          -DWarpX_PSATD=ON                  \
+          -DWarpX_PRECISION=SINGLE          \
+          -DWarpX_PARTICLE_PRECISION=SINGLE \
+          -DWarpX_QED_TABLE_GEN=ON
+
+        cmake --build build -j 2
+        ./build/bin/warpx.3d Examples/Physics_applications/laser_acceleration/inputs_3d
```
Comment (ax3l) [inline .github/workflows/ubuntu.yml:120]:
```suggestion
        ./build/bin/warpx.3d Examples/Physics_applications/laser_acceleration/inputs_3d
        ./build/bin/warpx.3d Examples/Physics_applications/laser_acceleration/inputs_rz
```

File: .github/workflows/ubuntu.yml [hunk 5]
```diff
@@ -80,6 +80,46 @@ jobs:
...
+        restore-keys: |
+          ccache-openmp-3D-RZ-SP-${{ hashFiles('.github/workflows/ubuntu.yml') }}-
+          ccache-openmp-3D-RZ-SP-
+    - name: build WarpX
+      run: |
+        cmake -S . -B build                 \
+          -GNinja                           \
+          -DCMAKE_VERBOSE_MAKEFILE=ON       \
+          -DWarpX_DIMS="RZ;3"                  \
+          -DWarpX_EB=OFF                    \
```
Comment (ax3l) [inline .github/workflows/ubuntu.yml]:
```suggestion
          -DWarpX_DIMS="RZ;3"               \
          -DWarpX_EB=OFF                    \
```

**Pull request merged**